### DISCUSSION TITLE
why does "select js->'$.field' as f from test where f = 'str' " work while "select js->'$.field' from test where js->'$field' = 'str'" does not ?

### DISCUSSION THREAD

**Discussion by littlegreenbean33** [duckdb/duckdb#6736] [Q&A]
I struggle with the subject problem. I thought I was doing something wrong.

I presume it is by design that you cannot reference the value of a json field in a where clause

ver 0.7.1 / python 3.10

----------------- first edit

OK I figureout one thing I seem to need ->>

still I can't make this work,

`
select
	ldate,
	ltime,
	msg,
	js,
	js->>'$.ip' as ip,
	js->'$.ip' as ip2,
    
from
	logs
where
	msg like '%DHCP%'
--and contains(ip,'192.168.3.195') 
--and ip like '%192.168.3.195%'
-- and ip = '192.168.3.195'
limit 10
`

--------------------------------  second edit

The above results happen using DBeaber 23.0.0.  and 7.0.1 drivers form linux ubuntu 20.04

However, I tried the same commands from python CLI and all works as expected.

could this be a bug either in JDBC drivers or DBeaber ?

**Comment by Mytherin** [comment]
The JDBC driver doesn't yet support nested types (see #6559), perhaps that is the issue you are running into?

**Reply by littlegreenbean33** [reply]
I am too newbie to make that statement.

I was not getting an error like the issue you mention. It was just silently not working.

If you consider this is related to #6559 please consider closing the question.

**Comment by lnkuiper** [comment]
What exactly does not work? You can reference a JSON field like so:
```sql
create table test as select '{"duck": 42}' js;
select * from test where js->>'duck' = '42';
┌──────────────┐
│      js      │
│   varchar    │
├──────────────┤
│ {"duck": 42} │
└──────────────┘
select js->>'duck' val from test where val = '42';
┌─────────┐
│   val   │
│ varchar │
├─────────┤
│ 42      │
└─────────┘
```

**Reply by littlegreenbean33** [reply]
it does using Python.

In my case/config it did not worked as expected using DBeaber

**Comment by littlegreenbean33** [comment]
Sorry  to bother you guys again with similar issue...or so I think.

This example works the same in CLI and python and return 7 results in my case.
```
select *,js->'$.DPT' as dpt from  logs where dpt = '"777"'; 
```

However this
```
select *,js->'$.DPT' as dpt from  logs where js->'$.DPT'  = '"777"';
```
returns an empty set.  I expected both queries would be equivalent. what is it I am missing ?

Thanks for your help.

JCG

**Reply by littlegreenbean33** [reply]
OK I found an answer   myself..... just in case it is useful for anybody else

accessing via ->> will return a VARCHAR and the comparison then works. 

Either wise -> will return a JSON...that needs to be compared against a JSON typed value.
